snorkel/labeling/analysis.py

```diff
@@ -354,7 +354,8 @@ def lf_summary(
         d["Conflicts"] = Series(data=self.lf_conflicts(), index=lf_names)
 
         if Y is not None:
-            confusions = [confusion_matrix(Y, self.L[:, i])[1:, 1:] for i in range(m)]
+            slice_abstain = [slice(int(x), None) for x in ((self.L == -1).sum(axis=0) > 0)]
```

Comment:
We may be able to simplify this by using the `labels` kwarg in scikit-learn's `confusion_matrix()` method. It let's you specify what the total label space is, so we can just make sure that it always includes -1 before we lop off the abstains for our calculations. Something like this:
```
labels = np.unique(np.concatenate((Y.flatten(), self.L.flatten(), np.array([-1]))))
confusions = [confusion_matrix(Y, self.L[:, i], labels)[1:, 1:] for i in range(m)]
```

Reply:
Good point, I'll do that. Thank you for the tips.